src/shared/lib/extendCBioPortalAPI.ts

```diff
@@ -0,0 +1,50 @@
+import CBioPortalAPI from 'shared/api/generated/CBioPortalAPI';
+import { SiteError } from 'AppStore';
+import { getBrowserWindow } from 'cbioportal-frontend-commons';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+export function extendCBioPortalAPI<T extends Constructor<{}>>(Base: T) {
+    return class extends Base {
+        handleErrorsGlobally(errorConfig: Partial<SiteError> = {}) {
+            const self = this;
+
+            // @ts-ignore (it's private, but we CAN access it)
+            const proxiedRequest = this.request;
+
+            function restoreRequest() {
+                // @ts-ignore (it's private, but we CAN access it)
+                self.request = proxiedRequest;
+            }
+
+            // @ts-ignore (it's private, but we CAN access it)
+            self.request = function() {
```

Comment:
so, superagent makes requests by invoking the client's instance of request.  I need to monkey with that request in order to insert an error handler.  But I only want to do this on an individual method call basis.  So, as soon as the call is made, I restore the request to what it was so that any future requests won't be affected.     I don't love this but after a lot of tinkering, it was the only way I could figure out to get the dev interface I wanted.  I could also have proxied all methods on the client and added another parameter, but it would half been harder i think from typescript standpoint.

Reply:
It might be possible to inject code using decorators instead. I haven't done much research on that stuff though, maybe it is not worth the effort.